packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.ts

```diff
@@ -420,6 +420,22 @@ export const useSpinButtonStyles_unstable = (state: SpinButtonState): SpinButton
     },
   });
 
+  let buttonClassName: string;
```

Comment:
This feels a bit heavy and I wonder if we couldn't simplify it a bit with something like:

```ts
// This is defined in the root of the module since it will never change
const appearanceMap: Record<string, string> = {
   outline: 'outline',
   //....
   'filled-lighter-shadow': 'filled-lighter',
   'filled-darker-shadow': 'filled-darker'
} as const;

// In useStyles where the classnames are merged
buttonStyles[appearanceMap[appearance]];
```

This would eliminate all the conditionals and make appearance styling more declarative, be less code overall (smaller bundle) and is probably faster to execute since it's just a lookup rather than several `if...elses`.

Reply:
not sure if this is an issue, but it would end up looking like this due to the type `Record<string,string>`

`buttonStyles[appearanceMap[appearance] as keyof typeof buttonStyles]`